### PR TITLE
Pca

### PR DESCRIPTION
**Pull request by dipak5053**
we have implemented all the modification according your email!

### CODE REVIEW

**Review by nmaarnio — CHANGES_REQUESTED** [review]
Thanks for making the modifications so fast. The code is already looking pretty good, but there were some style issues that I commented. Also, I think you accidentally pushed the MLP files to this pull request. Please remove them in the next commit you make to this open pull request, and store your MLP in a different branch/PR.

Regarding the tests, do you think it would be possible to make some tests where the result dataframe from the PCA is compared to some hard-coded dataframe that you know for sure has the correct/expected values from the PCA? These dataframes can be small so that hard-coding is possible. These kind of "visual tests" that you have are good, but they ultimately belong to a Jupyter notebook. You can find examples of Jupyter notebook testing in the repository and perhaps transfer your current test to this format.

**Comment by nmaarnio** [comment]
Also, please remember to run the linter @dipak5053 . If you have issues with it, I can also try to help. You can find the guide for it in [the contributing guide](https://github.com/GispoCoding/eis_toolkit/blob/master/CONTRIBUTING.md#code-style). We have started to use a tool **pre-commit** now. It will check that your coding style is correct when you are making commits, and you can run it manually/disable it also - all this is showed in the guide too. This will make it easier for you and other developers to adhere to the style requirements and reduce the PR reviewing time/work. Please note that before you can use pre-commit, you need to pull the latest additions (the configuration file needed for pre-commit to work) from the master branch. While in your current pca branch, you should be able to run
`git pull origin master`
and get the updates, after which you can activate pre-commit.

File: eis_toolkit/exceptions.py [hunk 1]
```diff
@@ -48,3 +48,11 @@ class CoordinatesOutOfBoundsException(Exception):
...
 
     pass
+
+class InvalidDataFrameException(Exception):
+    """Exception error class for invalid dataframes."""
+    pass
+
+class InvalidNumberOfPrincipalComponent(Exception):
+    """Exception error class for invalid dataframes."""
+    pass
```
[the remark this answers is not in the document]
Comment (zelioluca) [reply eis_toolkit/exceptions.py:58]: Hi I m luca! I m helping dipak in coding! Can you clarify this little bit?

[the remark this answers is not in the document]
Comment (zelioluca) [reply eis_toolkit/exceptions.py:58]: Ah I got it

**Comment by dipak5053** [comment]
Hi,
Hope you are doing well
I have fixed the issue.

Warm regards,
Dipak Kumar Nidhi

**Review by Mtk112 — APPROVED** [review]
Everything looks good to me now.

**Pull request merged**